src/libraries/System.IO.FileSystem/src/System/IO/FileSystem.Unix.cs

```diff
@@ -381,6 +381,17 @@ public static void MoveDirectory(string sourceFullPath, string destFullPath)
                 throw new IOException(SR.Format(SR.IO_AlreadyExists_Name, destFullPath));
             }
 
+#if TARGET_BROWSER
+            if (!Directory.Exists(Path.GetDirectoryName(sourceFullPath)))
```

Comment:
I'd add a comment which explains this is a workaround:
```suggestion
            // renaming a file doesn't return correct error code on emscripten if one of the parent paths does not exist,
            // manually workaround it for now (https://github.com/dotnet/runtime/issues/40305)
            if (!Directory.Exists(Path.GetDirectoryName(sourceFullPath)))
```